### PR TITLE
Bugfix/#237 task effort

### LINKED ISSUE #237: Change effort datatype from int to double
I find it odd that the effort for a task has to be an integer.
A task that takes 0.5 or 1.5 hours is a very likely scenario.

I think this should be changed!
What do you think @outcobra/developers?

### PR DESCRIPTION
**Pull request by needToRoll**
This closes #237 
Summary of changes:
- ​implemented DurationService
- added remaining effort to task details
- fixed effort rendering bug

I made sure to:
- [x] Add and / or update tests
- [x] Run the code formatter (`CTRL + ALT + L`)
- [x] Update any documentation (javadoc, domain model, erd, ...)
- [x] Follow the contribution guidelines
- [x] Update CI script and configs on Jenkins (`Jenkinsfile`, `auth0.yml`, `application.yml`, etc.)

Please review @jmesserli  / cc: @outcobra/developers.

### CODE REVIEW

**Comment by M4R1KU** [comment]
|`frontend/src/app/core/services/duration.service.ts:26`|
|---|
There's no need to create a variable

File: frontend/src/app/task/task-detail-component/task-detail.component.html [hunk 1]
```diff
@@ -32,6 +32,9 @@
                 </div>
             </div>
         </oc-icon-data>
+        <oc-icon-data iconName="timelapse" titleKey="i18n.modules.task.dialogs.add.effort.remaining">
+            <p class="task-property">{{getRemainingEffort()}}</p>
```
Comment (M4R1KU) [inline frontend/src/app/task/task-detail-component/task-detail.component.html]: There should be spaces between the curly braces and the statement (Could be wrong in other files too)

File: frontend/src/app/task/task-detail-component/task-detail.component.ts [hunk 2]
```diff
@@ -73,7 +83,6 @@ export class TaskDetailComponent implements OnInit, AfterViewInit {
         this._router.navigate(['/task']);
     }
```
Comment (M4R1KU) [inline frontend/src/app/task/task-detail-component/task-detail.component.ts]: |`line 71`|
|---|
Remove whitespace

**Comment by M4R1KU** [comment]
|`frontend/src/app/core/services/duration.service.ts:34`|
|---|
Move this method up and order them all of them in a logical way (seconds-minutes-hours or hours-minutes-seconds)

File: frontend/src/app/task/task-detail-component/task-detail.component.html (hunk 1 above)
Comment (M4R1KU) [inline frontend/src/app/task/task-detail-component/task-detail.component.html]: I'll do that myself

Comment (needToRoll) [inline frontend/src/app/task/task-detail-component/task-detail.component.html]: no I found a matching regex-pattern and replaced it already.

**Pull request merged**